src/cow-react/modules/swap/hooks/useSetupSwapState.ts

```diff
@@ -21,7 +21,8 @@ export function useSetupSwapState() {
   const dispatch = useAppDispatch()
   const tradeStateFromUrl = useTradeStateFromUrl()
   const swapState = useSwapState()
-  const wasAccountChanged = previousAccount && account && previousAccount !== account
+  const accountDisconnected = !account && !!previousAccount
+  const wasAccountChanged = accountDisconnected || (previousAccount && account && previousAccount !== account)
```

Comment:
Would it make sense to make a separate hook for this?
I think this is something that we might call in different places.

Reply:
The hooks must be called only once to setup Swap state.
It's responsible for only one task - get parameters from URL and localStorage and put them into the store.

Reply:
I meant the separate hook just for wasAccountChanged value, something like `useWasAccountChanged`.

Reply:
Oh, I see. Yes, it might be helpful